x/evm/keeper/state_transition.go

```diff
@@ -218,31 +218,43 @@ func (k *Keeper) ApplyUnsignedMessage(ctx sdk.Context, txHash common.Hash, msg e
 			res.Logs = types.NewLogsFromEth(receipt.Logs)
 			ctx.EventManager().EmitEvents(tmpCtx.EventManager().Events())
 		}
+	} else {
+		receipt.Status = ethtypes.ReceiptStatusFailed
 	}
 
 	// k.RefundGas doesn't work in omni's context, because we don't actually run the ante handler for each of these tx
 	// so the fee colelctor doesn't get the fees in the first place, and therefore doesn't have enough to refund what is due
 	// refund gas in order to match the Ethereum gas consumption instead of the default SDK one.
 	// if err = k.RefundGas(ctx, msg, msg.Gas()-res.GasUsed, cfg.Params.EvmDenom); err != nil {
-	// 	return nil, errorsmod.Wrapf(err, "failed to refund gas leftover gas to sender %s", msg.From())
+	// 	return nil, nil, errorsmod.Wrapf(err, "failed to refund gas leftover gas to sender %s", msg.From())
 	// }
 
 	if len(receipt.Logs) > 0 {
 		// Update transient block bloom filter
 		k.SetBlockBloomTransient(ctx, receipt.Bloom.Big())
 		k.SetLogSizeTransient(ctx, uint64(txConfig.LogIndex)+uint64(len(receipt.Logs)))
+	} else {
+		// logs cannot be empty
+		receipt.Logs = []*ethtypes.Log{}
```

Comment:
isnt this already handled by above line `logs := types.LogsToEthereum(res.Logs)`?

Reply:
This is mainly for the end block stuff which marshal the receipt causing some issues with the log field if it's set to empty.